lib/vtls/schannel_verify.c

```diff
@@ -77,21 +77,158 @@ static int is_cr_or_lf(char c)
   return c == '\r' || c == '\n';
 }
 
-static CURLcode add_certs_to_store(HCERTSTORE trust_store,
-                                   const char *ca_file,
-                                   struct Curl_easy *data)
+/* Search the substring needle,needlelen into string haystack,haystacklen
+ * Strings don't need to be terminated by a '\0'.
+ * Similar of OSX/Linux memmem (not available on Visual Studio).
+ * Return position of beginning of first occurence or NULL if not found
+ */
+static const char *c_memmem(const void *haystack, size_t haystacklen,
+                            const void *needle, size_t needlelen)
+{
+  const char *p;
+  char first;
+  const char *str_limit = (const char *)haystack + haystacklen;
+  if(((const char *)haystack + needlelen) > str_limit)
+    return NULL;
+  if(needlelen == 0)
+    return NULL;
```

Comment:
I would rewrite this as `!needlelen || needlelen > haystacklen` rather than check against str_limit

Reply:
done